src/main/controllers/ChallengeController.kt

```diff
@@ -77,7 +60,7 @@ class ChallengeController: DefaultController<Challenge>(), RestController<Challe
             finalResult.data = serviceResult.data
         }
 
-        if (finalResult.result != SOAResultType.SUCCESS && finalResult.message != null) {
+        if (finalResult.result != SOAResultType.SUCCESS && finalResult.message!!.contains("You do not have enough valid shares available")) {
```

Comment:
do finalResult.message?.contains

this can throw an error if its and error type without a message